src/fu-util.c

```diff
@@ -2382,6 +2382,63 @@ fu_util_check_polkit_actions (GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_generate_motd (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	g_autoptr(GPtrArray) devices = NULL;
+	guint upgrade_count = 0;
+	g_autofree gchar *upgrade_str = NULL;
+
+	/* check args */
+	if (g_strv_length (values) != 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_ARGS,
+				     "Invalid arguments");
+		return FALSE;
+	}
+
+	/* get devices from daemon */
+	devices = fwupd_client_get_devices (priv->client, NULL, error);
+	if (devices == NULL)
+		return FALSE;
+	for (guint i = 0; i < devices->len; i++) {
+		FwupdDevice *dev = g_ptr_array_index (devices, i);
+		g_autoptr(GPtrArray) rels = NULL;
+		g_autoptr(GError) error_local = NULL;
+
+		/* not going to have results, so save a D-Bus round-trip */
+		if (!fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_SUPPORTED))
+			continue;
+
+		/* get the releases for this device and filter for validity */
+		rels = fwupd_client_get_upgrades (priv->client,
+						  fwupd_device_get_id (dev),
+						  NULL, &error_local);
+		if (rels == NULL)
+			continue;
+		upgrade_count++;
+	}
+	if (upgrade_count == 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOTHING_TO_DO,
+				     _("No firmware upgrades available"));
+		return FALSE;
+	}
+
+	/* TRANSLATORS: Display a summary message for the number of updates available and
+	   how to run them  */
+	upgrade_str = g_strdup_printf (ngettext ("%u device has a firmware upgrade available. "
+						 "Run `fwupdmgr get-upgrades` for more information",
```

Comment:
Isn't `get-updates` the non-compat command? I forget, so ignore me otherwise.

Reply:
I don't remember which came first.  The message will only come in this version of fwupd or newer, so I don't think it actually matters which is shown.

Reply:
True.